base/interceptors.go

```diff
@@ -117,3 +120,41 @@ func (as authStreamingClientConn) Receive(req any) error {
 
 	return as.StreamingClientConn.Receive(req)
 }
+
+type circuitBreakerInterceptor struct {
+	cb failsafe.Executor[connect.AnyResponse]
+}
+
+func newCircuitBreakerInterceptor() *circuitBreakerInterceptor {
+	circuitBreaker := circuitbreaker.Builder[connect.AnyResponse]().
+		WithFailureThresholdRatio(6, 10).
+		Build()
+
+	return &circuitBreakerInterceptor{
+		cb: failsafe.NewExecutor(circuitBreaker),
```

Comment:
I think there are some failures that shouldn't trip the circuit breaker. For example, the store RPCs that include error details... we shouldn't prevent further requests if there have been some operations discarded or validations failed.

I reckon this needs to only consider a few error codes that indicate unexpected failure (`unknown`, `unavailable`, `internal` spring to mind).